Player.gd

```diff
@@ -94,7 +94,14 @@ func get_input(delta):
 	elif velocity.x == 0:
 		anim = "idle"
 	
-	sprite.play(anim + "_" + dir)
+	if(
+		anim == "idle" &&
+		sprite.animation == "fire_" + dir &&
+		fire_cooldown_timer > 0
+	):
+		pass
```

Comment:
couldn't you just check the opposite?

Reply:
I considered it. But I felt this was more readable at the time. It was late.

Reply:
Well it works and we're not being judged on code quality - at least I don't think so - it is a Github hack!
I'll merge it and if you feel like tidying it up on another PR then go ahead :)